### PR TITLE
[helm] nhc

### PR DESCRIPTION
**Pull request by rustielin**
### Description

Get an installation of Node Health Checker in k8s testnets. In conjunction with Chaos Mesh, we should be able to use this setup to validate individual node performance across various simulated network conditions.

To install node health checker, use the Terraform variable `var.enable_node_health_checker`, and configure it using `var.node_health_checker_helm_values`. Configuration values include the URL for a baseline node as well as a mint key. In the testnet setup, we use `validator-0` as the baseline node, and we can run NHC on the rest of the nodes.

### Test Plan

Apply to rustienet. In my deployment, I'm using `validator-0` as the baseline node, and giving NHC 15.5 vCPU and 24Gi memory.

The default service is not publicly available, but we can access it via port-forward:

```
$ kubectl port-forward svc/node-health-checker 20121:20121
```

Inject some latency in the direction of: `NHC --> validator-1`

```
kind: NetworkChaos
apiVersion: chaos-mesh.org/v1alpha1
metadata:
  namespace: default
  name: latency-val1-to-nhc-300ms
spec:
  selector:
    namespaces:
      - default
    labelSelectors:
      app.kubernetes.io/instance: validator-1
  mode: all
  action: delay
  delay:
    latency: 300ms
    correlation: '0'
    jitter: 0ms
  direction: to
  target:
    selector:
      namespaces:
        - default
      labelSelectors:
        app.kubernetes.io/name: node-health-checker
    mode: all
```

Invoke NHC:
```
$ curl http://localhost:20121/check_node\?node_url\=http://aptos-node-1-validator:8080\&baseline_configuration_name\=ait3_registration
```

Results show latency around 600ms, which is what we expect after injecting 300ms one-way latency:

```
{"evaluation_results":[{"headline":"Chain ID reported by baseline and target match","score":100,"explanation":"The node under investigation reported the same Chain ID TESTING as is reported by the baseline node","evaluator_name":"node_identity","category":"api","links":[]},{"headline":"Role Type reported by baseline and target match","score":100,"explanation":"The node under investigation reported the same Role Type validator as is reported by the baseline node","evaluator_name":"node_identity","category":"api","links":[]},{"headline":"Proposals count is increasing","score":100,"explanation":"Successfully pulled metrics from target node twice and saw that proposals count is increasing (from 72018 to 72140)","evaluator_name":"proposals","category":"consensus","links":[]},{"headline":"Consensus round is increasing","score":100,"explanation":"Successfully pulled metrics from target node twice and saw that consensus round increased (from 143898 to 144130)","evaluator_name":"round","category":"consensus","links":[]},{"headline":"Consensus timeouts metric okay","score":100,"explanation":"The consensus timeouts count was 19 in the first round and 19 in the second round of metrics collection, which is within tolerance of the allowed increase (0)","evaluator_name":"timeouts","category":"consensus","links":[]},{"headline":"State sync version is within tolerance","score":100,"explanation":"Successfully pulled metrics from target node twice, saw the version was progressing, and saw that it is within tolerance of the baseline node. Target version: 437964. Baseline version: 438064. Tolerance: 5000.","evaluator_name":"version","category":"state_sync","links":[]},{"headline":"Average API latency is good","score":100,"explanation":"The average API latency was 603ms, which is below the maximum allowed latency of 1000ms. Note, this latency is not the same as standard ping latency, see the attached link.","evaluator_name":"latency","category":"api","links":["https://aptos.dev/nodes/node-health-checker-faq#how-does-the-latency-evaluator-work"]},{"headline":"Transaction processing speed is sufficient","score":100,"explanation":"The minimum TPS (transactions per second) required of nodes is 1000, your node hit: 1503 (out of 1503 transactions submitted per second). Your node could theoretically hit even higher TPS, the evaluation suite only tests to check your node meets the minimum requirements.","evaluator_name":"tps","category":"performance","links":[]},{"headline":"Target node produced valid recent transaction","score":100,"explanation":"We were able to pull the same transaction (version: 437418) from both your node and the baseline node. Great! This implies that your node is keeping up with other nodes in the network.","evaluator_name":"transaction_availability","category":"api","links":[]}],"summary_score":100,"summary_explanation":"100: Awesome!"}%
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3030)

### CODE REVIEW

**Comment by sherry-x** [comment]
I like the idea, could you please make a separate helm chart for NHC? I think many community folks express the interest of running this along with their node as well

File: terraform/helm/aptos-node/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -44,6 +44,19 @@ spec:
...
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: node-health-checker
+    ports:
+    - protocol: TCP
+      port: 9101
+    - protocol: TCP
+      port: 8080
+    - protocol: TCP
+      port: 6180
```
Comment (sherry-x) [inline terraform/helm/aptos-node/templates/networkpolicy.yaml:59]: what type of check is running on 6180?

**Review by sherry-x — APPROVED** [review]

File: terraform/helm/aptos-node/templates/networkpolicy.yaml (hunk 1 above)
Comment (rustielin) [reply terraform/helm/aptos-node/templates/networkpolicy.yaml:59]: @banool ? We require the node's noise port in configuration, but I'm not sure what checks are actually using that: https://github.com/aptos-labs/aptos-core/blob/b183a232784e4c77991b23e8728d3c7669a95d47/ecosystem/node-checker/configuration_examples/single_node_validator.yaml

Comment (banool) [reply terraform/helm/aptos-node/templates/networkpolicy.yaml:59]: Yeah they're not, but that's still a potential development route, where we do noise based checks.

**Pull request merged**